chunk/tags.go

```diff
@@ -25,34 +25,35 @@ import (
 	"time"
 
 	"github.com/ethersphere/swarm/sctx"
+	"github.com/ethersphere/swarm/spancontext"
 )
 
 // Tags hold tag information indexed by a unique random uint32
 type Tags struct {
 	tags *sync.Map
-	rng  *rand.Rand
 }
 
 // NewTags creates a tags object
 func NewTags() *Tags {
 	return &Tags{
 		tags: &sync.Map{},
-		rng:  rand.New(rand.NewSource(time.Now().Unix())),
 	}
 }
 
 // Create creates a new tag, stores it by the name and returns it
 // it returns an error if the tag with this name already exists
 func (ts *Tags) Create(s string, total int64) (*Tag, error) {
 	t := &Tag{
-		Uid:       ts.rng.Uint32(),
+		Uid:       uint32(rand.Int31()),
 		Name:      s,
 		StartedAt: time.Now(),
 		Total:     total,
 	}
 	if _, loaded := ts.tags.LoadOrStore(t.Uid, t); loaded {
 		return nil, errExists
 	}
+
+	t.ctx, t.span = spancontext.StartSpan(context.Background(), "new.upload.tag")
```

Comment:
Here also, to allow the context propagation (by accepting it as the argument to Create), instead creating a new one.